sdk/src/main/scala/com/horizen/account/state/StateTransition.scala

```diff
@@ -41,6 +43,8 @@ class StateTransition(view: AccountStateView, messageProcessors: Seq[MessageProc
             throw err
           // any other exception will bubble up and invalidate the block
         } finally {
+          // make sure we disable automatic gas consumption in case a message processor enabled it
+          view.disableGasTracking()
```

Comment:
so we move it here to avoid `refundGas` consuming another gas, right?

Reply:
This was moved here to make sure automatic gas consumption is only active while the message processor is executing. Afterwards we don't want it anymore. Before this change it could cause an `OutOfGas` exception while doing `addBalance` in `refundGas`. Doesn't matter really, because for [EVM-274](https://horizenlabs.atlassian.net/browse/EVM-274?atlOrigin=eyJpIjoiZmVhZjdkZjYwYWU3NDcwOWFhYzdiMDg2NDFjOGNhNGIiLCJwIjoiaiJ9) this will be removed.